### PR TITLE
Add autonomy messages and associated converters

### PR DESCRIPTION
**Pull request by kylestach**
## Description
Add messages required for autonomy modules (planning, control, radio), not including gameplay.

Marked as draft because this needs documentation.

## Associated Issue
Issue #1579 

## Design Documents
N/A

### [PENDING] Automated tests

Expected result: Conversions are lossless

### CODE REVIEW

File: soccer/src/soccer/gameplay/robocup-py.cpp [hunk 1]
```diff
@@ -185,8 +185,15 @@ void our_robot_add_local_obstacle(OurRobot* self, rj_geometry::Shape* obs) {
     self->local_obstacles(shared_obs);
 }
 
-void our_robot_set_avoid_ball_radius(OurRobot* self, float radius) {
-    self->avoid_ball_radius(radius);
+void our_robot_set_avoid_ball_radius([[maybe_unused]] OurRobot* self,
+                                     [[maybe_unused]] float radius) {
+    // TODO(Kyle): Use SPDLog here
```
Comment (oswinso) [inline soccer/src/soccer/gameplay/robocup-py.cpp]: issue number?

File: soccer/src/soccer/radio/packet_convert.cpp [hunk 2]
```diff
@@ -8,6 +8,9 @@
 #include "robot_status.hpp"
 #include "motion/motion_setpoint.hpp"
 
+// TODO(Kyle): Make this a real ROS parameter
```
Comment (oswinso) [inline soccer/src/soccer/radio/packet_convert.cpp]: Issue number?

File: soccer/src/soccer/robot.cpp [hunk 3]
```diff
@@ -14,13 +14,16 @@ using namespace rj_geometry;
...
 using Planning::MotionCommand;
 
-/** thresholds for avoidance of opponents - either a normal (large) or an
- * approach (small)*/
-constexpr float kOppAvoidSmallThresh = 0.03;
-constexpr float kOppAvoidSmall = kRobotRadius - kOppAvoidSmallThresh;
-/** threshold for avoiding the ball*/
-constexpr float kBallAvoidSmallMult = 2.0;
-constexpr float kBallAvoidSmall = kBallAvoidSmallMult * kBallRadius;
+// TODO(Kyle): Make these ROS parameters and move them to a central location
```
Comment (oswinso) [inline soccer/src/soccer/robot.cpp:17]: issue number?

File: soccer/src/soccer/robot.cpp [hunk 4]
```diff
@@ -353,94 +326,6 @@ void OurRobot::face(rj_geometry::Point pt) {
...
-            FieldDimensions::current_dimensions.center_radius());
-    }
-
-    // create an obstacle if necessary
-    if (intent().avoid_ball_radius > 0.0) {
-        return std::make_shared<rj_geometry::Circle>(context_->world_state.ball.position,
-                                                    intent().avoid_ball_radius);
-    }
-    return nullptr;
-}
```
Comment (oswinso) [inline soccer/src/soccer/robot.cpp:443]: Is this removed because it is unused?

**Review by oswinso — CHANGES_REQUESTED** [review]
Just issue numbers and one question.

File: soccer/src/soccer/robot.cpp (hunk 4 above)
Comment (kylestach) [reply soccer/src/soccer/robot.cpp:443]: It's been unused for a while but this PR also removed the corresponding useless items from `RobotIntent`.

File: rj_constants/include/rj_constants/topic_names.hpp [hunk 5]
```diff
@@ -32,3 +34,56 @@ constexpr auto kDetectionFramePub = "vision_receiver/detection_frame";
...
+
+}  // namespace planning::topics
+
+namespace control {
+
+namespace topics {
+
+static inline std::string manipulator_setpoint_pub(int robot_id) {
+    return "robot/" + std::to_string(robot_id) + "/control/manipulator_setpoint";
+}
```
Comment (oswinso) [inline rj_constants/include/rj_constants/topic_names.hpp:63]: What's manipulator? We have manipulators?

File: soccer/src/soccer/gameplay/robocup-py.cpp (hunk 1 above)
Comment (oswinso) [reply soccer/src/soccer/gameplay/robocup-py.cpp]: bump

File: soccer/src/soccer/joystick/manual_control_node.cpp [hunk 6]
```diff
@@ -90,12 +90,15 @@ void ManualControlNode::update_intent_and_setpoint(OurRobot* robot) {
                                       ? RobotIntent::TriggerMode::ON_BREAK_BEAM
                                       : RobotIntent::TriggerMode::IMMEDIATE)
                                : RobotIntent::TriggerMode::STAND_DOWN;
-    intent.kcstrength = controls_.kick_power;
+
+    // TODO(Kyle): The controller should directly work in the (0, max_kick_speed] range
```
Comment (oswinso) [inline soccer/src/soccer/joystick/manual_control_node.cpp]: Another TODO. issue number?

**Review by oswinso — CHANGES_REQUESTED** [review]
Issue number + questions.

File: rj_constants/include/rj_constants/topic_names.hpp (hunk 5 above)
Comment (kylestach) [reply rj_constants/include/rj_constants/topic_names.hpp:63]: Manipulator = everything that's not the drivetrain.

I didn't have a better name for it.

Comment (oswinso) [reply rj_constants/include/rj_constants/topic_names.hpp:63]: so like the kicker / chipper? Or are there more things here?

**Review by oswinso — APPROVED** [review]
lgtm

**Review by HussainGynai — APPROVED** [review]
LGTM

**Pull request merged**